packages/playwright-core/src/server/webkit/wkPage.ts

```diff
@@ -856,7 +857,20 @@ export class WKPage implements PageDelegate {
     const omitDeviceScaleFactor = scale === 'css';
     this.validateScreenshotDimension(rect.width, omitDeviceScaleFactor);
     this.validateScreenshotDimension(rect.height, omitDeviceScaleFactor);
-    const result = await this._session.send('Page.snapshotRect', { ...rect, coordinateSystem: documentRect ? 'Page' : 'Viewport', omitDeviceScaleFactor });
+
+    let result: any;
+    // Take screenshot in the UI process where possible. This way we capture CSS transformations.
+    const takeSnapshotInUIProcess = os.platform() === 'linux' ||
```

Comment:
`os.platform()` does not work for remote case, we should get the actual platform of the browser. We already do that in `crBrowser` somehow.

Alternatively, browser should figure our the best strategy and either `Playwright.takePageScreenshot` or forward to `Page.snapshotRect`.